census/src/main/scala/hmda/census/model/CbsaLookup.scala

```diff
@@ -34,8 +34,10 @@ object CbsaLookup extends CbsaResourceUtils {
     }
   }.toSeq
 
-  val codeMap: Map[String, String] = {
-    values.map(x => (x.cbsa, x.cbsaTitle)).toMap
+  def nameFor(cbsaId: String): String = {
+    values.find(_.cbsa == cbsaId)
```

Comment:
Just an improvement in modularity of code and separation of concerns. Don't know about performance.